aries_cloudagent/resolver/base.py

```diff
@@ -38,6 +39,20 @@ class ResolverType(Enum):
     NON_NATIVE = "non-native"
 
 
+class ResolutionResult:
+    """Resolution Class to pack the DID Doc and the resolution information."""
+
+    def __init__(self, did_doc: DIDDocument, resolver_metadata: dict):
```

Comment:
Should we formalize the `resolver_metadata`? We've already kind of outlined what it contains by what is returned in it in the resolve method; perhaps we should make it a `namedtuple`?